ndindex/integer.py

```diff
@@ -98,3 +98,23 @@ def newshape(self, shape):
         self.reduce(shape)
 
         return shape[1:]
+
+    def as_subindex(self, index):
+        from .ndindex import ndindex
+        from .slice import Slice
+        from .tuple import Tuple
```

Comment:
Is there a reason to do these lazily?

Reply:
Probably some subset of imports could be moved to the top-level, but if we move them all, we get circular import issues. If you have a suggestion on how to better deal with this let me know.

Reply:
You could import the module, rather than the class. `from . import tuple`, then use `tuple.Tuple` instead, or even set `Tuple = tuple.Tuple` in the methods here

Reply:
I'm not sure if that would work, but unfortunately my submodules shape names with built-ins, so I won't want to import them directly. Also `ellipsis` is the same name for the submodule and item which causes its own problems.

Reply:
I think I can fix it by moving the imports to the bottom of the file. But I prefer to fix circular import issues by moving imports inside of functions rather than to the bottom of the file.

Reply:
I think some of these module names should change.